src/test/java/com/puppycrawl/tools/checkstyle/xpath/ElementNodeTest.java

```diff
@@ -58,6 +59,61 @@ public void init() throws Exception {
         rootNode = new RootNode(rootAst);
     }
 
+    @Test
+    public void testNodeOrdering() throws Exception {
+        final String xpath = "//METHOD_DEF[./IDENT[@text='callSomeMethod']]/SLIST/*";
+        final List<NodeInfo> nodes = getXpathItems(xpath, rootNode);
+        final boolean result = nodes
```

Comment:
I executed this test with different `compareOrder` method implementation, it always passes, it is not ok

As far as I understand this `depth` approach, it should be fully correlate with line/column ordering. 
It means for all nodes in `getXpathItems(xpath, rootNode);` the following condition is true:
`node[i].lineNumber < node[j].lineNumber or (node[i].lineNumber = node[j].lineNumber and node[i].columnNumber < node[j].columnNumber)` where *i < j*.
It is easier to write because all `NodeInfo` subclasses implement `getLineNumber(), getColumnNumber()` using real `DetailAst` values.

There is more difficult way to write this test:
- get results from `getXpathItems(xpath, rootNode);`
- convert them to `DetailAST[]` using `convertToArray(List<NodeInfo> nodes)` method
- get `rootAst` from `RootNode` 
- traverse `rootAst` with DFS and check that all elements `DetailAST[]` appear in exact same order

Reply:
Okay, I'll make the changes